zkevm-circuits/src/bytecode_circuit/bytecode_unroller.rs

```diff
@@ -688,6 +707,47 @@ fn into_words(message: &[u8]) -> Vec<u64> {
     words
 }
 
+/// BytecodeCircuit
+#[derive(Clone, Default, Debug)]
+pub struct BytecodeCircuit<F: Field> {
+    /// Unrolled bytecodes
+    pub bytecodes: Vec<UnrolledBytecode<F>>,
+    /// Circuit size
+    pub size: usize,
+}
+
+impl<F: Field> BytecodeCircuit<F> {
+    /// new BytecodeCircuitTester
+    pub fn new(bytecodes: Vec<UnrolledBytecode<F>>, size: usize) -> Self {
+        BytecodeCircuit { bytecodes, size }
+    }
+}
+
+impl<F: Field> SubCircuit<F> for BytecodeCircuit<F> {
+    type Config = BytecodeCircuitConfig<F>;
+
+    fn new_from_block(block: &witness::Block<F>) -> Self {
+        let bytecode_size = block.circuits_params.max_bytecode + 128;
```

Comment:
I checked this `+128` number and it's to account for the unusable rows that halo2 needs.  It's not a parameter that the consumer should choose, and it can be calculated from the circuit configuration. Currently it shows up here as a magic number.  I agree that this should be addressed, although I would prefer to do it in a future PR to avoid adding more stuff here!
https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/949

Reply:
Sounds good to me 👍